tests/lib/rules/no-pause-test.js

```diff
@@ -0,0 +1,129 @@
+/**
+ * @fileoverview Disallow usage of pauseTest() in tests
+ * @author Scott Kidder
+ */
+'use strict';
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+let rule = require('../../../lib/rules/no-pause-test'),
+  RuleTester = require('eslint').RuleTester;
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+const TEST_FILE_NAME = 'some-test.js';
+
+let ruleTester = new RuleTester({
+  parserOptions: {
+    ecmaVersion: 2018,
+    sourceType: 'module',
+  },
+});
+
+ruleTester.run('no-pause-test', rule, {
+  valid: [
+    {
+      filename: TEST_FILE_NAME,
+      code: `
+        import setupApplicationTest from 'ember-qunit';
+        import { module, test } from 'qunit';
+
+        module('Acceptance | module', function(hooks) {
+          setupApplicationTest(hooks);
+
+          test('foo', async function(assert) {
+            assert.dom('.foo').hasText('Bar');
+          });
+        });
+      `,
+    },
+    {
+      filename: TEST_FILE_NAME,
+      code: `
+        import setupApplicationTest from 'ember-qunit';
+        import { module, test } from 'qunit';
+        import { pauseTest } from '@ember/test-helpers';
+
+        module('Acceptance | module', function(hooks) {
+          setupApplicationTest(hooks);
+
+          test('foo', async function(assert) {
+            // await pauseTest();
```

Comment:
It's not helpful to add a test case for commented out code. I would remove this test case.